diff -urN meson-0.46.1/mesonbuild/modules/gnome.py meson-0.46.1.new/mesonbuild/modules/gnome.py
--- meson-0.46.1/mesonbuild/modules/gnome.py	2018-05-16 18:23:56.000000000 +0200
+++ meson-0.46.1.new/mesonbuild/modules/gnome.py	2018-06-19 22:19:00.491036568 +0200
@@ -854,7 +854,7 @@
         modulename = args[0]
         if not isinstance(modulename, str):
             raise MesonException('Argument must be a string')
-        return ModuleReturnValue(os.path.join('share/gtk-doc/html', modulename), [])
+        return ModuleReturnValue(os.path.join('share/doc/gtk-doc/html', modulename), [])
 
     @staticmethod
     def _unpack_args(arg, kwarg_name, kwargs, expend_file_state=None):
diff -urN meson-0.46.1/mesonbuild/scripts/gtkdochelper.py meson-0.46.1.new/mesonbuild/scripts/gtkdochelper.py
--- meson-0.46.1/mesonbuild/scripts/gtkdochelper.py	2018-05-16 18:23:56.000000000 +0200
+++ meson-0.46.1.new/mesonbuild/scripts/gtkdochelper.py	2018-06-19 22:18:27.554479460 +0200
@@ -244,7 +244,7 @@
         install_gtkdoc(options.builddir,
                        options.subdir,
                        install_prefix,
-                       'share/gtk-doc/html',
+                       'share/doc/gtk-doc/html',
                        install_dir)
     return 0
 
