--- meson-1.8.1/data/macros.meson.orig	2025-05-23 23:51:05.000000000 +0200
+++ meson-1.8.1/data/macros.meson	2025-05-25 18:34:27.941441519 +0200
@@ -1,6 +1,6 @@
 %__meson %{_bindir}/meson
 %__meson_wrap_mode nodownload
-%__meson_auto_features enabled
+%__meson_auto_features auto
 %__meson_verbose 1
 
 %meson \
@@ -22,13 +22,14 @@
         --sharedstatedir=%{_sharedstatedir} \
         --wrap-mode=%{__meson_wrap_mode} \
         --auto-features=%{__meson_auto_features} \
+        --default-library=both \
         %{_vpath_srcdir} %{_vpath_builddir} \
         %{nil}}
 
 %meson_build \
     %{shrink:%{__meson} compile \
         -C %{_vpath_builddir} \
-        -j %{_smp_build_ncpus} \
+        %{?_smp_mflags}
         %{?__meson_verbose:--verbose} \
         %{nil}}
 
@@ -42,7 +43,7 @@
 %meson_test \
     %{shrink:%{__meson} test \
         -C %{_vpath_builddir} \
-        --num-processes %{_smp_build_ncpus} \
+        %{?_smp_mflags} \
         --print-errorlogs \
         %{nil}}
 
