--- meson-1.6.1/data/macros.meson.orig	2024-12-16 21:34:40.000000000 +0100
+++ meson-1.6.1/data/macros.meson	2025-02-06 21:07:08.380379917 +0100
@@ -1,6 +1,6 @@
 %__meson %{_bindir}/meson
 %__meson_wrap_mode nodownload
-%__meson_auto_features enabled
+%__meson_auto_features auto
 %__meson_verbose 1
 
 %meson \
@@ -22,13 +22,14 @@
         --sharedstatedir=%{_sharedstatedir} \
         --wrap-mode=%{__meson_wrap_mode} \
         --auto-features=%{__meson_auto_features} \
+        --default-library=both \
         %{_vpath_srcdir} %{_vpath_builddir} \
         %{nil}}
 
 %meson_build \
     %{shrink:%{__meson} compile \
         -C %{_vpath_builddir} \
-        -j %{_smp_build_ncpus} \
+        %{?_smp_mflags}
         %{?__meson_verbose:--verbose} \
         %{nil}}
 
@@ -42,6 +43,6 @@
 %meson_test \
     %{shrink:%{__meson} test \
         -C %{_vpath_builddir} \
-        --num-processes %{_smp_build_ncpus} \
+        %{?_smp_mflags}
         --print-errorlogs \
         %{nil}}
